src/core/thread/topology.hpp

```diff
@@ -800,7 +800,7 @@ class Neighbor : public InstanceLocatorInit
 #endif
     uint8_t         mVersion;  ///< The MLE version
     LinkQualityInfo mLinkInfo; ///< Link quality info (contains average RSS, link margin and link quality)
-#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE || OPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE
```

Comment:
needed in both unfortunatelly

Reply:
I agree with @edmont. As I see, an Initiator doesn't need `mLinkMetricsSeriesInfoList`. I think after we seperate `GetEnhAckProbingMetrics` and `SetEnhAckProbingMetrics` from other methods, it can be built successfully. Thoughts? @canisLupus1313